lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -206,6 +211,50 @@ protected function _getPortableTableColumnDefinition($tableColumn)
         return $column;
     }
 
+
+    /**
+     * Return Doctrine/Mysql-compatible column default values for MariaDB 10.2.7+ servers.
+     *
+     * - Since MariaDb 10.2.7 column defaults stored in information_schema are now quoted
+     *   to distinguish them from expressions (see MDEV-10134 for a what is an expression).
+     * - CURRENT_TIMESTAMP, CURRENT_TIME, CURRENT_DATE are stored in information_schema
+     *   as current_timestamp(), currdate(), currtime()
+     * - Literal escaping is normalized in information schema (store "''" instead of "\'")
+     * - Note: columnDefault should only be null when column is not_nullable otherwise
+     *   a string 'NULL' should be given. Unfortunately, it's not 100% correct in case of
+     *   upgrade from previous versions: see https://jira.mariadb.org/browse/MDEV-14053.
+     *   Otherwise we could simplify the condition === 'NULL' || === null by adding the parameter
+     *   is_nullable.
+     *
+     * @link https://mariadb.com/kb/en/library/information-schema-columns-table/
+     * @link https://jira.mariadb.org/browse/MDEV-13132
+     * @link https://jira.mariadb.org/browse/MDEV-14053
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(MariaDb102Platform $platform, ?string $columnDefault) : ?string {
+
+        if ($columnDefault === 'NULL' || $columnDefault === null) {
+            return null;
+        }
+        if (strpos($columnDefault, "'") === 0) {
```

Comment:
Wouldn't `$columnDefault[0] === "'"` be better here?

Reply:
Old habit... always found easier to search and replace by `mb_strpos` when needed. In this case the index can be used safely. I fixed.